libtelemetry/src/test/java/com/mapbox/android/telemetry/SchemaTest.java

```diff
@@ -0,0 +1,528 @@
+import com.google.gson.JsonPrimitive;
+import com.google.gson.annotations.SerializedName;
+
+import org.json.JSONException;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.zip.GZIPInputStream;
+
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+public class SchemaTest {
+  private static final String SCHEMA_URL = "https://mapbox.s3.amazonaws.com/mapbox-gl-native/event-schema/mobile-event-schemas.jsonl.gz";
+  private static final String APP_USER_TURNSTILE = "appUserTurnstile";
+  private static final String LOCATION = "location";
+  private static final String MAP_CLICK = "map.click";
+  private static final String MAP_DRAG = "map.dragend";
+  private static final String MAP_LOAD = "map.load";
+  private static final String NAVIGATION_ARRIVE = "navigation.arrive";
+  private static final String NAVIGATION_CANCEL = "navigation.cancel";
+  private static final String NAVIGATION_DEPART = "navigation.depart";
+  private static final String NAVIGATION_FASTER_ROUTE = "navigation.fasterRoute";
+  private static final String NAVIGATION_FEEDBACK = "navigation.feedback";
+  private static final String NAVIGATION_OVERLAPPING_AUDIO = "navigation.overlappingAudio";
+  private static final String NAVIGATION_REROUTE = "navigation.reroute";
+  private static final String NAVIGATION_SIGNIFICANT_SPEED_DIFFERENCE = "navigation.significantSpeedDifference";
+  private static final String NAVIGATION_TURNSTILE = "navigation.turnstile";
+  private static final String VISIT = "visit";
+  private ArrayList<JsonObject> schemaArray;
+
+
+  @Before
+  public void downloadSchema() throws Exception {
+    final CountDownLatch latch = new CountDownLatch(1);
+    final AtomicBoolean failureRef = new AtomicBoolean();
+    Callback callback = provideACallback(latch, failureRef);
+
+    Request request = new Request.Builder()
+      .url(SCHEMA_URL)
+      .build();
+
+    OkHttpClient client = new OkHttpClient();
+    client.newCall(request).enqueue(callback);
+
+    latch.await();
+  }
+
+  @Test
+  public void checkAppUserTurnstileEvent() throws Exception {
+    JsonObject schema = grabSchema(APP_USER_TURNSTILE);
+    List<Field> fields = grabClassFields(AppUserTurnstile.class);
+
+    System.out.println("schema: " + schema);
+    System.out.println("fields: " + fields);
+
+    assertEquals(schema.size(), fields.size());
```

Comment:
Generally is preferable having only 1 assertion per test 👀 https://speakerdeck.com/guardiola31337/elegant-unit-testing-mobilization-2016?slide=57
If this tests fails we won't know if it was because of an error with the fields size or because the `schemaContainsFields` assertion. What about creating different tests for each check?
The same thing happens with the rest of the tests 👇